app/models/ems_refresh/save_inventory_helper.rb

```diff
@@ -113,10 +113,31 @@ def save_child_inventory(obj, hashes, child_keys, *args)
 
   def store_ids_for_new_records(records, hashes, keys)
     keys = Array(keys)
-    hashes.each do |h|
-      r = records.detect { |r| keys.all? { |k| r.send(k) == r.class.type_for_attribute(k.to_s).cast(h[k]) } }
```

Comment:
yeah I was thinking the casted values would be loaded up front for the index, giving an array of values.  Then you just look them up by that same Array from the record.